TestApp/Sources/Reader/Common/ReaderViewController.swift

```diff
@@ -216,6 +224,84 @@ class ReaderViewController: UIViewController, Loggable {
         searchViewController = vc
     }
     
+    // MARK: - Highlights
+    
+    private func addHighlightDecorationsObserverOnce() {
+        if let decorator = self.navigator as? DecorableNavigator {
+            decorator.observeDecorationInteractions(inGroup: highlightDecorationGroup) { event in
+                self.activateDecoration(event)
+            }
+        }
+    }
+    
+    private func updateDecorationForAnyNewHighlight() {
```

Comment:
This will also update the decorations when modifying an existing highlight (e.g. color).

```suggestion
    private func updateHighlightDecorations() {
```